libs/parallelism/algorithms/include/hpx/parallel/algorithms/swap_ranges.hpp

```diff
@@ -117,17 +94,201 @@ namespace hpx { namespace parallel { inline namespace v1 {
+                auto dist2 = detail::distance(first2, last2);
+                return parallel_swap_ranges(std::forward<ExPolicy>(policy),
+                    first1, first2, dist1 < dist2 ? dist1 : dist2);
+            }
+        };
+        /// \endcond
+    }    // namespace detail
+
+    template <typename ExPolicy, typename FwdIter1, typename FwdIter2>
+    HPX_DEPRECATED_V(1, 8,
+        "hpx::parallel::transform_exclusive_scan is deprecated, use "
+        "hpx::transform_exclusive_scan instead")
     inline typename std::enable_if<hpx::is_execution_policy<ExPolicy>::value,
         typename util::detail::algorithm_result<ExPolicy, FwdIter2>::type>::type
-    swap_ranges(
-        ExPolicy&& policy, FwdIter1 first1, FwdIter1 last1, FwdIter2 first2)
+        swap_ranges(
+            ExPolicy&& policy, FwdIter1 first1, FwdIter1 last1, FwdIter2 first2)
     {
         static_assert((hpx::traits::is_forward_iterator<FwdIter1>::value),
             "Requires at least forward iterator.");
         static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
             "Requires at least forward iterator.");
 
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
         return detail::swap_ranges<FwdIter2>().call(
             std::forward<ExPolicy>(policy), first1, last1, first2);
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::swap_ranges
+    HPX_INLINE_CONSTEXPR_VARIABLE struct swap_ranges_t final
+      : hpx::functional::tag_fallback<swap_ranges_t>
+    {
+        // clang-format off
+        template <typename FwdIter1, typename FwdIter2,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
+                hpx::traits::is_forward_iterator<FwdIter2>::value
+            )>
+        // clang-format on
+        friend FwdIter2 tag_fallback_dispatch(hpx::swap_ranges_t,
+            FwdIter1 first1, FwdIter1 last1, FwdIter2 first2)
+        {
+            static_assert(hpx::traits::is_input_iterator<FwdIter1>::value,
+                "Requires at least forward iterator.");
+            static_assert(hpx::traits::is_output_iterator<FwdIter2>::value,
+                "Requires at least forward iterator.");
+
+            return hpx::parallel::v1::detail::swap_ranges<FwdIter2>().call(
+                hpx::execution::seq, first1, last1, first2);
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
```

Comment:
If you check for `is_forward_iterator` here, the `static_assert` below is superfluous (and btw wrong in this case anyways).